apps/vmq_server/test/vmq_publish_SUITE.erl

```diff
@@ -833,6 +834,34 @@ forward_properties(_Config) ->
     disable_on_publish(),
     disable_on_subscribe().
 
+max_packet_size(Config) ->
+    enable_on_publish(),
+    enable_on_subscribe(),
+    SubClientId = vmq_cth:ustr(Config) ++ "-sub",
+    PubClientId = vmq_cth:ustr(Config) ++ "-pub",
+    Topic = list_to_binary(vmq_cth:utopic(Config)),
+    Pub = packetv5:gen_publish(Topic, 0, <<"publish">>, [{properties,
+                                                          #{p_user_property => [{<<"hello">>, <<"world">>}]}}]),
+    ReducedPub = packetv5:gen_publish(Topic, 0, <<"publish">>, []),
+    SubConnect = packetv5:gen_connect(SubClientId,
+                                      [{keepalive, 60}, {properties, #{p_max_packet_size => iolist_size(ReducedPub)}}]),
+    SubConnack = packetv5:gen_connack(0, ?M5_CONNACK_ACCEPT, #{}),
+    {ok, SubSocket} = packetv5:do_client_connect(SubConnect, SubConnack, []),
+    Subscribe = packetv5:gen_subscribe(77, [packetv5:gen_subtopic(Topic,0)], #{}),
+    ok = gen_tcp:send(SubSocket, Subscribe),
+    SubAck = packetv5:gen_suback(77, [0], #{}),
+    ok = packetv5:expect_frame(SubSocket, SubAck),
+    PubConnect = packetv5:gen_connect(PubClientId, [{keepalive, 60}]),
+    PubConnack = packetv5:gen_connack(0, ?M5_CONNACK_ACCEPT, #{}),
+    {ok, PubSocket} = packetv5:do_client_connect(PubConnect, PubConnack, []),
+    ok = gen_tcp:send(PubSocket, Pub),
+    ok = packetv5:expect_frame(SubSocket, ReducedPub),
+    ok = gen_tcp:close(PubSocket),
+    ok = gen_tcp:close(SubSocket),
+    disable_on_publish(),
+    disable_on_subscribe(),
+    ok.
+
```

Comment:
It looks like dropping messages hasn't been implemented? `Where a Packet is too large to send, the Server MUST discard it without sending it and then behave as if
it had completed sending that Application Message [MQTT-3.1.2-25].`